Add 'Managed by Puppet' template headers

baurmatt opened the pull request:
Fixes #594.

baurmatt commented:
Any update on this? This lays around for over 2 months now :)

mmoll commented on templates/server/config.ru.erb:
```diff
@@ -1,5 +1,7 @@
-## Module:           '<%= scope.to_hash['module_name'] %>'
+#
+# WARNING: This file is auto-generated by puppet!
+# WARNING: To change it, edit the file in puppet/templates/server/config.ru.erb
```
I'm really unsure if it's a good idea to tell users to edit files... They should rather set parameters to chnage the generated files, IMHO.

baurmatt replied:
Good point, i've removed that line.

mmoll commented:
This also should get rebased, there are some more files in `templates/server/puppetserver/conf.d/` now.

baurmatt commented:
All done :)

mmoll commented:
While personally I don't like these headers, this can go in, IMHO.

ekohl commented:
I don't like these headers either. Especially the `Warning` part. In the context of our installer I'm not sure it really is a warning.

hdep commented:
IMHO Those header are very usefull and almost a standard for puppet. In one shot you see that the file has been modified by puppet or not.

'Warning' or not isn't a big deal. Could be just  : 

'File managed by puppet'

baurmatt commented:
I'm happy to change the wording, just tell we what it's supposed to say! :)

ekohl commented:
I'd just stick with `Managed by puppet` as we already have in some places.

baurmatt commented:
Done :)

ekohl commented on spec/classes/puppet_server_config_spec.rb:
```diff
@@ -201,7 +201,6 @@
 
         it 'should contain autosign.conf with out content set' do
           should contain_file("#{confdir}/autosign.conf")
-          should_not contain_file("#{confdir}/autosign.conf").with_content(/# Managed by Puppet/)
```
Could you keep this in but modified to the current header?

baurmatt replied:
Done

baurmatt commented:
* Rebased from master
* Removed exclamation mark
* Couldn't find any templates which get joined together by the following search criteria:
  * erb templates using template() function
  * concat::fragment's using template() function
  * template() function output gets concated with another template() function output

mmoll commented:
@ekohl do we want to get this in?

baurmatt commented:
(Hope this ends up on Github)

Will fix the merge conflict on monday morning :)

Send from one of my mobile devices

Am 11.01.2019 um 19:57 schrieb Ewoud Kohl van Wijngaarden <notifications@github.com <mailto:notifications@github.com> >:

@ekohl approved this pull request.

I guess this is OK, but there is now a merge conflict. One in the tests due to newlines, the other because we removed config.ru <http://config.ru> . Given I'm about to release I might fix it up myself on Monday if it hasn't by then.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.

ekohl commented:
I fixed it up and merged it in f4e008e1546c2ecf091cb379766fc9e3f0148350. Thanks!
